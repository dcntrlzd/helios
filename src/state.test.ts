import * as fs from 'fs';
import * as path from 'path';

import State from './state';

const STATE_PATH = path.resolve(__dirname, '../test/state.json');

afterEach(() => {
  if (fs.existsSync(STATE_PATH)) {
    fs.unlinkSync(STATE_PATH);
  }
});

it('can create a new state instance', () => {
  const state = new State({ path: STATE_PATH });
  expect(fs.readFileSync(STATE_PATH).toString()).toMatchSnapshot();
});

it('does not persists if no state path is given', () => {
  const state = new State({ path: false });
  expect(fs.existsSync(STATE_PATH)).toBe(false);
  state.setState(420, { version: 0 })
  expect(fs.existsSync(STATE_PATH)).toBe(false);
  expect(state.getState(420)).toMatchSnapshot();
});

it('can persist state modifications', async () => {
  const state = new State({ path: STATE_PATH });
  expect(fs.readFileSync(STATE_PATH).toString()).toMatchSnapshot();
  state.setState(420, { version: 0 });
  expect(fs.readFileSync(STATE_PATH).toString()).toMatchSnapshot();
  expect(state.getState(420)).toMatchSnapshot();
});

it('can preload a previously persisted state', () => {
  fs.writeFileSync(STATE_PATH, JSON.stringify({ 420: { version: 0 } }));
  const state = new State({ path: STATE_PATH });
  expect(state.getState(420)).toMatchSnapshot();
});
