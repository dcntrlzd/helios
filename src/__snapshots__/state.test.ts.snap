// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can create a new state instance 1`] = `"{}"`;

exports[`can persist state modifications 1`] = `"{}"`;

exports[`can persist state modifications 2`] = `"{\\"420\\":{\\"version\\":0}}"`;

exports[`can persist state modifications 3`] = `
Object {
  "version": 0,
}
`;

exports[`can preload a previously persisted state 1`] = `
Object {
  "version": 0,
}
`;

exports[`does not persists if no state path is given 1`] = `
Object {
  "version": 0,
}
`;
